src/Symfony/Component/HttpKernel/DataCollector/DataCollector.php

```diff
@@ -100,4 +106,24 @@ protected function varToString($var)
 
         return $this->valueExporter->exportValue($var);
     }
+
+    /**
+     * @return callable[] The casters to add to the cloner
+     */
+    protected function getCasters()
+    {
+        return array(
+            '*' => function ($v, array $a, Stub $s, $isNested) {
```

Comment:
should be the main fix: nested objects are truncated, which prevent deep recursion into nested structures by default (eg "subjects" of the security decision log, requests attributes, etc.)